roc/log.go

```diff
@@ -181,6 +192,36 @@ func loggerRoutine() {
 	}
 }
 
+func getGID() uint64 {
+	b := make([]byte, 64)
+	b = b[:runtime.Stack(b, false)]
+	b = bytes.TrimPrefix(b, []byte("goroutine "))
+	b = b[:bytes.IndexByte(b, ' ')]
+	n, _ := strconv.ParseUint(string(b), 10, 64)
+	return n
+}
```

Comment:
LogMessage TID field is for thread ID, not goroutine ID. I think it's better to keep filling it with TID, not GID, for two reasons:

* to avoid confusion when reading logs, otherwise you don't know who to interpret the integer you see if you don't know if the message originated from C or Go

* to keep track of native threads; we may see in logs what native threads are used in C code, and it may be important to understand if they are the same or different native threads used in Go code (e.g. when a Go callback is invoked from C)

Reply:
I get errors flagged for this 
```
./log.go:6:1: error: unknown type name 'uint64_t'
uint64_t rocGoThreadID();
```

```
go list failed to return CompiledGoFiles. This may indicate failure to perform cgo processing; try building at the command line. See https://golang.org/issue/38990.go list
```

Reply:
This is just a prototype, I didn't test the code.

Reply:
i'm not quite sure how to go about this

Reply:
No worries, then let's keep TID unset for now, and I'll implement the C part by myself after merging this PR.

Reply:
alright noted